tests/acceptance/features/webUIUpload/upload.feature

```diff
@@ -16,19 +16,21 @@ Feature: File Upload
     When the user uploads file "new-lorem.txt" using the webUI
     Then no message should be displayed on the webUI
     And file "new-lorem.txt" should be listed on the webUI
-    And as "Alice" the content of "new-lorem.txt" should be the same as the content of local file "new-lorem.txt"
+    And as "Alice" the content of "new-lorem.txt" in the server should be the same as the content of local file "new-lorem.txt"
 
   @smokeTest @ocisSmokeTest
   Scenario: simple upload of a folder that does not exist before
-    When the user uploads a folder containing the following files in separate sub-folders using the webUI:
+    Given a folder "CUSTOM" has been created with the following files in separate sub-folders in the server
```

Comment:
I don't understand why the words "in separate sub-folders" are in this step. All the files are in the same folder, in this case a folder called "CUSTOM".

"in the server" is a bit odd here. The files seem to actually be created "by the middleware" and are then visible to the test-runner to use as the source of upload.

But we could refactor these words in a separate PR later.

Reply:
@phil-davis the given step has to be fixed in the middleware. It was working the same as the step description before but using session ids as folder names. We can store every file provided to some sub-folder like `sub1`, `sub2` according to the order, or even provide table data with sub-folder names.

Reply:
At the moment that step seems to create each file directly in the specified folder, there are no sub-folders created.

"or even provide table data with sub-folder names" - that sounds good.

We can refactor in the middleware in the future to have a step like:
```
Given a folder "CUSTOM" has been created with the following files and folders in the middleware
  | folder | file |
  |  | topFile.txt |
  | sub1 | someFile.txt |
  | sub1 | anotherFile.txt |
  | sub2/sub3 | file.txt |
  | sub4 |  |
  | sub5/sub6/sub7 |  |
```

If the folder column is empty, then the file goes directly in the main folder.
If the file column is empty, then the folder(s) are created, with the bottom folder empty
If the folder column has a multiple-folder path then create that whole folder tree.

IMO that will make a flexible step for creating whole file-folder trees in the middleware that can be used in a scenario.